### PR TITLE
Fix bugs reported by mercury#5

### PR DESCRIPTION
**Pull request by rubio41**
- [x] Chinese bug for last Monday VSO Bug ID 2314
- [x] "Afterter" bug VSO Bug ID 2339
- [x] "morning" and "tomorrow morning" will be extracted together VSO Bug ID 2376
- [x] "the week" "my week" "the day" VSO Bug ID 2377, 2378
- [x] filter the ambiguous single month mention VSO Bug ID 2389

### CODE REVIEW

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishMergedExtractorConfiguration.cs [hunk 1]
```diff
@@ -10,10 +10,16 @@ public class SpanishMergedExtractorConfiguration : IMergedExtractorConfiguration
         public static readonly Regex AfterRegex = 
             new Regex(@"(despues(\s*de(\s+las?)?)?)", RegexOptions.IgnoreCase | RegexOptions.Singleline);
 
-        //TODO: change this to Spanish if there is same requirement of split from A to B as two time points
+        //TODO: change the following three regexes to Spanish if there is same requirement of split from A to B as two time points
         public static readonly Regex FromToRegex = 
             new Regex(@"\b(from).+(to)\b.+", RegexOptions.IgnoreCase | RegexOptions.Singleline);
 
+        public static readonly Regex SingleAmbiguousMonthRegex =
+            new Regex(@"\b(may|march)\b", RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishMergedExtractorConfiguration.cs]: In following patches, please always put the regexes in the YAML files. We can add notes for which ones may be language-specific.

File: Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedExtractor.cs [hunk 2]
```diff
@@ -82,6 +98,19 @@ private void AddTo(List<ExtractResult> dst, List<ExtractResult> src)
             return config.FromToRegex.IsMatch(er.Text);
         }
 
+        private bool FilterAmbiguousSingleWord(ExtractResult er, string text)
+        {
+            if (config.SingleAmbiguousMonthRegex.IsMatch(er.Text.ToLowerInvariant()))
+            {
+                var stringBefore = text.Substring(0, (int) er.Start).TrimEnd();
+                if (!config.PrepositionSuffixRegex.IsMatch(stringBefore))
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedExtractor.cs]: There should be more negative unit tests to be sure this filtering doesn't break common scenarios.

File: Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs [hunk 3]
```diff
@@ -19,7 +19,7 @@ public static class DateTimeDefinitions
 	{
 		public const string TillRegex = @"(?<till>to|till|until|thru|through|--|-|—|——)";
 		public const string RangeConnectorRegex = @"(?<and>and|through|to|--|-|—|——)";
-		public const string RelativeRegex = @"(?<order>next|upcoming|this|last|past|previous|current|over the)";
```
Comment (tellarin) [inline Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: What happened to the "over the" case?

File: Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs [hunk 4]
```diff
@@ -19,7 +19,7 @@ public static class DateTimeDefinitions
 	{
 		public const string TillRegex = @"(?<till>to|till|until|thru|through|--|-|—|——)";
 		public const string RangeConnectorRegex = @"(?<and>and|through|to|--|-|—|——)";
-		public const string RelativeRegex = @"(?<order>next|upcoming|this|last|past|previous|current|over the)";
+		public const string RelativeRegex = @"(?<order>next|upcoming|this|last|past|previous|current|the|my)";
```
Comment (tellarin) [inline Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: Will "my" now be extracted? Or that will come later?

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishMergedExtractorConfiguration.cs (hunk 1 above)
Comment (rubio41) [reply Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishMergedExtractorConfiguration.cs]: there is no YAML file for Spanish right now. I don't want to reference the EnglishDatetimeDefinition in this class.

File: Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs (hunk 3 above)
Comment (rubio41) [reply Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: We will extract "the"now. "over" is similar as "around", "at" which should not be extracted.

File: Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs (hunk 4 above)
Comment (rubio41) [reply Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: I include "my" for now to make it the same as "the" and "this".

**Pull request merged**